builder/vsphere/common/step_add_flag.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type FlagConfig
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-sdk/template/interpolate"
+	"github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/driver"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type FlagConfig struct {
+	// Enable Virtualization Based Security option for virtual machine. Defaults to `false`.
+	// Requires `vvtd_enabled` and `NestedHV` to be set to `true`.
+	// Requires `vTPM` to be set to `true`.
+	// Requires `firmware` to be set to `efi-secure`.
+	VbsEnabled bool `mapstructure:"vbs_enabled"`
+	// Enable IO/MMU option for virtual machine. Defaults to `false`.
+	VvtdEnabled bool `mapstructure:"vvtd_enabled"`
+}
+
+func (c *FlagConfig) Prepare(h *HardwareConfig) ([]string, []error) {
+	var errs []error
+	var strings []string
+
+	if c.VbsEnabled {
+		if !c.VvtdEnabled {
+			errs = append(errs, fmt.Errorf("`vvtd_enabled` must be set to `true` when `vbs_enabled` is set to `true`"))
+		}
+
+		if !h.NestedHV {
+			errs = append(errs, fmt.Errorf("`nestedhv` must be set to `true` when `vbs_enabled` is set to `true`"))
+		}
+
+		if !h.VTPMEnabled {
+			errs = append(errs, fmt.Errorf("`vtpm` must be set to `true` when `vbs_enabled` is set to `true`"))
+		}
+
+		if h.Firmware != "efi-secure" {
+			errs = append(errs, fmt.Errorf("`firmware` must be set to `efi-secure` when `vbs_enabled` is set to `true`"))
+		}
+	}
+
+	return strings, errs
+}
+
+type StepAddFlag struct {
+	FlagConfig FlagConfig
+	Ctx        interpolate.Context
```

Comment:
Is this context used for anything?

Reply:
Will review. I think I have `FlagConfig.Prepare` is using it.

Reply:
Addressed in https://github.com/hashicorp/packer-plugin-vsphere/commit/440ec1f12c2d1b30cd6422eb475a8bd8e8807437.